src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -359,50 +376,170 @@ internal async Task<IEnumerable<JObject>> GetFunctionTriggers(IEnumerable<Functi
+
+            string durableMajorVersion;
             if (FileUtility.FileExists(hostJsonPath))
             {
-                var json = JObject.Parse(await FileUtility.ReadAsync(hostJsonPath));
+                var hostJson = JObject.Parse(await FileUtility.ReadAsync(hostJsonPath));
+
+                durableMajorVersion = await GetDurableMajorVersionAsync(hostJson, hostOptions);
 
                 // get the DurableTask extension config section
                 JToken extensionsValue;
-                if (json.TryGetValue("extensions", StringComparison.OrdinalIgnoreCase, out extensionsValue) && extensionsValue != null)
+                if (hostJson.TryGetValue("extensions", StringComparison.OrdinalIgnoreCase, out extensionsValue) && (extensionsValue as JObject) != null)
                 {
-                    json = (JObject)extensionsValue;
+                    hostJson = (JObject)extensionsValue;
                 }
 
                 // we will allow case insensitivity given it is likely user hand edited
                 // see https://github.com/Azure/azure-functions-durable-extension/issues/111
                 JToken durableTaskValue;
-                if (json.TryGetValue(DurableTask, StringComparison.OrdinalIgnoreCase, out durableTaskValue) && durableTaskValue != null)
+                if (hostJson.TryGetValue(DurableTask, StringComparison.OrdinalIgnoreCase, out durableTaskValue) && durableTaskValue != null)
                 {
-                    try
+                    var durableHostConfig = (JObject)durableTaskValue;
+                    if (durableMajorVersion == null || durableMajorVersion.Equals("1"))
                     {
-                        var kvp = (JObject)durableTaskValue;
-                        if (kvp.TryGetValue(HubName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
-                        {
-                            config.Add(HubName, nameValue.ToString());
-                        }
-
-                        if (kvp.TryGetValue(DurableTaskStorageConnectionName, StringComparison.OrdinalIgnoreCase, out nameValue) && nameValue != null)
-                        {
-                            config.Add(Connection, nameValue.ToString());
-                        }
+                        return this.GetDurableV1Config(durableHostConfig);
                     }
-                    catch (Exception)
+                    else if (durableMajorVersion.Equals("2"))
+                    {
+                        return this.GetDurableV2Config(durableHostConfig);
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Invalid host.json configuration for version {durableMajorVersion}.x of durable task extension");
+                    }
+                }
+            }
+            else
+            {
+                durableMajorVersion = await GetDurableMajorVersionAsync(null, hostOptions);
```

Comment:
You're passing a null in here, but this function doesn't handle a null - it'll null ref? I don't see why you need this extra call here at all - can't you just have the single call above and have it do the right thing?

Reply:
This should be resolved with the suggestions I took from you.